Added docs to the page permissions routes (#153)
port-labs/port-docs: docs/software-catalog/page/page.md

---
sidebar_position: 1
---

# Page

> ## What is a Page?

Reviewer: ```suggestion
## What is a page?
```

<center>

![Save view menu button marked](../../../static/img/software-catalog/pages/SaveViewDropMenuButton.png)

</center>

After clicking the `Save as a new page` button, a window will pop up:

![Save as a new page popup](../../../static/img/software-catalog/pages/SaveAPageForm.png)

:::note
When saving a new page or editing an existing one, a set of icons is available to you:

<center>

![Page Icons dropdown menu](../../../static/img/software-catalog/pages/PageIcons.png)

</center>
:::

### Edit or delete a page

#### Editing pages

By clicking the `...` button on the top right corner, you can edit the page name and its icon.

<center>

![Page menu](../../../static/img/software-catalog/pages/PageMenu.png)

</center>

Editing a page:

![Edit Page popup window](../../../static/img/software-catalog/pages/EditPageForm.png)

#### Deleting pages

:::note
An Entity page is deleted automatically when the connected Entity is deleted.
:::

## Table operations

Table operations are used to define the user’s view of the Port platform.

:::tip
We highly recommend using the table operations to provide a clean and accurate view of the platform for your developers.

:::

> All table operations are available on the top bar of the table:

Reviewer: Hasn't this been written already in line 70?

Reviewer 2: Line 70 refers to page operations, this line refers to table operations

![Table operations bar](../../../static/img/software-catalog/pages/TableOperationsBar.png)

### Filter

> Filtering a table is done using the following menu:

Reviewer: ```suggestion
YOu can filter a table by using the following menu:
```

![Table filter menu marked](../../../static/img/software-catalog/pages/TableFilterMenu.png)

> You can define any filtering operator with a suitable value, in order to filter anything.
> 
> You can filter according to one or many fields while setting the relation between each field filter: `And/Or`.

Reviewer: ```suggestion
To filter, you can define any filtering operator with a suitable value.

You can filter one or more values while setting the relation between each field filter with a: `And/Or`.
```

### Sort

> Sorting the table is done on the following menu:

Reviewer: ```suggestion
You can sort the table on the following menu:
```

![Table sort menu marked](../../../static/img/software-catalog/pages/TableSortMenu.png)

You can sort by one or more fields of any kind.

:::tip
Sorting a specific column can be done by clicking on the column title.
:::

### Hide

> Hiding table columns is done on the following menu:

Reviewer: ```suggestion
You can hide table columns on the following menu:
```

![Table hide menu marked](../../../static/img/software-catalog/pages/TableHideMenu.png)

You can decide whether each field is viewable to users or not.

:::tip
We highly recommend hiding irrelevant data from users, in order to provide them a clean work environment with few irrelevant distractions.
:::

### Group By

> Grouping by Entities is done on the following menu:

Reviewer: ```suggestion
You can group by Entities on the following menu:
```

![Table group by menu marked](../../../static/img/software-catalog/pages/TableGroupByMenu.png)

You can group results according to any field in the table.

:::tip

> Group by is recommended when you want to create custom views for users, such as Microservices by Owners.

Reviewer: ```suggestion
Group by is recommended when you want to create custom views for users, such as "microservices by owners".
```

Just create your `group by` setting, add additional viewing settings if needed, and [save a new page](#saving-new-pages) from the custom view.
:::

### Search

> We provide a free text search option on tables. Searching on a table is available on the left side of the top bar:

Reviewer: ```suggestion
We provide a free text search option on tables. To search on a table, click the left side of the top bar:
```
